### PR TITLE
Update Sparse_matrix_Operation.py

### PR DESCRIPTION
**Pull request by Yash202000**
In this code, I added the addition and subtraction program code.
I hope this might helpful.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Yash202000** [comment]
Here in this code I added the addition and subtraction program with simple algorithm.

**Comment by Yash202000** [comment]
here in this code I added addition program which will store sorted output in list and display it.
hope this might helpful!!

**Comment by Yash202000** [comment]
I am part of DWOC

File: Python/ds/Sparse_matrix_Operation.py [hunk 1]
```diff
@@ -1,115 +1,165 @@
...
-
-    Given matrix is a sparse matrix
-    [0, 0, 2]
-    [0, 1, 1]
-    [1, 1, 2]
-
-
-Time Complexity:O(noOfRows*noOfcolumns)
-Space Complexity:O(1)
+1 0 0
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py:11]: Add description of the program at the top in commented form.
Add sample output, space and time complexity too and shift this doc string to the end of the program.

File: Python/ds/Sparse_matrix_Operation.py [hunk 2]
```diff
@@ -1,115 +1,165 @@
...
-                Print_Matrix[2][k] = matrix[i][j]
-                k += 1
-
-    for i in Print_Matrix:
-        print(i)
-
-
-if(count == size):
-    print("Null Mtrix")
+# this read_mat function will read input from user and return matrix as list of inputs
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py:38]: Do not leave blank lines between comment and code.

File: Python/ds/Sparse_matrix_Operation.py [hunk 3]
```diff
@@ -1,115 +1,165 @@
...
+        print(sp3)
+    else:
+        print("not possible")
+
+
+# substraction of sparse matrix program
+
+
+def substractspm(sp1, sp2):
+    print("performing substraction on given sparse matrices : ")
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py]: Remove this line.

File: Python/ds/Sparse_matrix_Operation.py [hunk 4]
```diff
@@ -1,115 +1,165 @@
...
+        for j in range(len(k[0])):
+            print(k[i][j], end=" ")
+        print()
+
+
+# addition function
+
+
+def addspm(sp1, sp2):
+    print("performing addition on given sparse matrices......... ")
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py]: Remove this line.

File: Python/ds/Sparse_matrix_Operation.py [hunk 5]
```diff
@@ -1,115 +1,165 @@
...
+            sp3.append([sp2[j][0], sp2[j][1], sp2[j][2]])
+            count += 1
+            j += 1
+        sp3.insert(0, [sp1[0][0], len(sp3[1]), len(sp3)])
+        display_matrix(sp3)
+    else:
+        return "addition operation is not possible."
+
+
+# trying to enhance my aproach from above code.
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py:111]: Can you please add some more comments to better explain your approach?

File: Python/ds/Sparse_matrix_Operation.py [hunk 6]
```diff
@@ -1,115 +1,165 @@
...
+                    j += 1
+        print(sp3)
+    else:
+        print("not possible")
+
+
+# substraction of sparse matrix program
+
+
+def substractspm(sp1, sp2):
```
Comment (atarax665) [inline Python/ds/Sparse_matrix_Operation.py:138]: Add this function in the driver code.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/ds/Sparse_matrix_Operation.py (hunk 6 above)
Comment (Yash202000) [reply Python/ds/Sparse_matrix_Operation.py:138]: https://github.com/TesseractCoding/NeoAlgo/commit/55f2531f4b33cc8eb153d3fda6636bede17f648c

I made some changes as you descripted. plz review it once.

**Comment by Yash202000** [comment]
plz review the changes made in code
commit id :- https://github.com/TesseractCoding/NeoAlgo/commit/55f2531f4b33cc8eb153d3fda6636bede17f648c

**Review by atarax665 — COMMENTED** [review]
Remove all the extra blank lines. Format your code according to PEP8 rules. You can use [this](http://pep8online.com/).

**Comment by HarshCasper** [comment]
@Yash202000 Please make the README updates as well, and once you are done, squash your commits. We can review it then, because having a lot of commits makes review hard for us.

**Comment by HarshCasper** [comment]
Closing the PR because of no updates!

**Pull request closed without merging**